test/e2e/kubernetes/pod/pod.go

```diff
@@ -559,24 +560,28 @@ func WaitOnReady(podPrefix, namespace string, successesNeeded int, sleep, timeou
 	for {
 		select {
 		case result := <-ch:
-			mostRecentWaitOnReadyErr = result.err
-			if result.ready {
-				successCount++
-				if successCount >= successesNeeded {
-					return true, nil
-				}
-			} else {
-				if successCount > 1 {
-					failureCount++
-					if failureCount >= successesNeeded {
-						PrintPodsLogs(podPrefix, namespace)
-						return false, errors.Errorf("Pods from deployment (%s) in namespace (%s) have been checked out as all Ready %d times, but NotReady %d times. This behavior may mean it is in a crashloop", podPrefix, namespace, successCount, failureCount)
+			mostRecentWaitOnSuccessesErr = result.err
+			if mostRecentWaitOnSuccessesErr == nil {
```

Comment:
Total nit, but might be good to invert the if statement to reduce nesting. For example:

```go
if mostRecentWaitOnSuccessesErr != nil {
  continue
}
// do the rest of the things without the if nesting
```

Reply:
This is interesting. I propose we bikeshed this sort of abstractly (at later time), and then apply the team-approved pattern generally. Because this pattern is the one we currently use to handle channel responses in these goroutine validations.